### ISSUE TITLE
torch.repeat is divergent from np.repeat

### ISSUE THREAD

**Issue by mruberry** [pytorch/pytorch#50013]
[torch.repeat](https://pytorch.org/docs/master/tensors.html?highlight=repeat#torch.Tensor.repeat), as its documentation notes, is divergent from [np.repeat](https://numpy.org/doc/stable/reference/generated/numpy.repeat.html?highlight=repeat#numpy.repeat) but similar to [np.tile](https://numpy.org/doc/stable/reference/generated/numpy.tile.html?highlight=tile#numpy.tile). Now that [torch.tile](https://pytorch.org/docs/master/generated/torch.tile.html?highlight=tile#torch.tile) is implemented, we can deprecate torch.repeat in favor of torch.til by doing the following:

- [ ] deprecate torch.repeat in favor of torch.tile for a release
  - [ ] verify torch.tile implements the same behavior as np.tile and torch.repeat
  - [ ] audit and update internal callsites so PyTorch doesn't throw warnings by default
- [ ] remove torch.repeat for a release (possibly requires writing an upgrade for serialized torchscript)
- [ ] restore torch.repeat analogous to np.repeat

**Comment by mruberry** [comment]
@Ed suggests adding an example of how old code using this behavior would be modified to use this new behavior. 

What would the release notes be?

**Comment by Hitansh159** [comment]
yes, currently tile and repeat do the same thing the only difference is that repeat will not except the input tuple of lesser dimensions then tensor. Tile overcome this by assuming that  user want to extend the dimensions to initial if input tuple is of lesser dimension. 

also Tile is implemented with help of repeat function.


I want to contribute but the issue being that I am unable to build at my side.
I am trying since last 24 hrs.
if you can guide me with it it would be really helpful.
I tried searching for it I found that it will need VS 2019 MSCV toolkit I downloaded that and tried build but the processes is going on since more than 7 hrs.
I don't know what is the issue.

**Comment by mruberry** [comment]
Hi @Hitansh159, thanks for your interest, but this issue is currently only for discussion. We have not decided to do any work here, yet. It'd be great to have you contribute to PyTorch, though. You may want to look at this query of issues for something to start with:

https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22

and check out our contribution guide here: https://pytorch.org/docs/stable/community/contribution_guide.html.

I'm sorry that you're having a problem building PyTorch. I'm not familiar with building on Windows, but if you're experiencing an issue then you may want to look at the PyTorch Forums (https://discuss.pytorch.org/) or ask a new question there, or if you believe you're encountering a bug in PyTorch on Windows then you can file an issue here describing the failure you're seeing.
